### ISSUE TITLE
Mention user (@)

### ISSUE THREAD

**Issue by Cu8eR** [flarum/core#1021]
For me it is not working correctly - official flarum forum

- press @ in editor and type for example "davis" - nobody is shown in a dropdown menu (user exists). 

Some ideas:
- add a user avatar next to the name in a dropdown menu

**Comment by tobscure** [comment]
Thanks for the report. I can't reproduce so can you please provide more specific instructions on how you're encountering this bug. e.g. which editor are you using (new discussion, reply?) which topic? does it happen after you reload the page? with other usernames? what about if you backspace a few times and then try again?

User avatar is already present next to the name in the dropdown menu.

**Comment by jordanjay29** [comment]
Also, what browser and OS are you using? Cannot reproduce on desktop Firefox or Chrome.

**Comment by kulga** [comment]
I have experienced this. After some investigating (Devflarum - dev-master) Firefox 48
it appears it happens after several lines
![](http://i.imgur.com/hO1c7oV.gif)

**Comment by Cu8eR** [comment]
Google Chrome last beta version (Win 10, 64 bit, Pro - anniversary update)

As @kulga - soemthing like this, but it happens only for some users

**Comment by clarkwinkelmann** [comment]
> @Eldenroot user exists

Only users already loaded from the API are suggested by the dropdown. The list of users will be far from complete.

https://github.com/flarum/flarum-ext-mentions/blob/9da2ac1e122f6c2f9dce8c75dd407219cce9372c/js/forum/src/addComposerAutocomplete.js#L111

In order to see a user he must have an interaction in the visible portion of the discussion, be visible on the homepage if you went from here, or appeared on any other page visited before starting to write the mention.

This doesn't explain @kulga 's demo though :sweat_smile:

**Comment by tobscure** [comment]
It will also hit the API for users as you type https://github.com/flarum/flarum-ext-mentions/blob/9da2ac1e122f6c2f9dce8c75dd407219cce9372c/js/forum/src/addComposerAutocomplete.js#L179

**Comment by tobscure** [comment]
I reckon @kulga's find is to do with dodgy positioning of the list, related to the scroll position of the textarea

**Comment by clarkwinkelmann** [comment]
Missed that API call when reading the code :see_no_evil: 

Yes, same issue happened to me with the emoji autocomplete when writing long messages. Not sure if that's reported as another issue.

**Comment by franzliedke** [comment]
IMO we should close this, until we have reproducible instructions for hitting the bug.

**Comment by tobscure** [comment]
1. Open post composer
2. Press enter a few dozen times
3. Type `@` and see absence of dropdown (it is positioned off the screen)

**Comment by franzliedke** [comment]
Oh, the original issue description made it sound like the dropdown appears, but without correct matches for usernames. Is that a different issue then?

**Comment by tobscure** [comment]
The original reported issue was not specific but is likely the same as what's happening in @kulga's comment. There's nothing to indicate that the username matching isn't working properly.

**Comment by franzliedke** [comment]
I suppose we calculate the positioning logic based on the position of the cursor within the textarea, while ignoring whether it has been scrolled down.
